src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -729,12 +738,16 @@ REDHAWK_PALEXPORT UInt32 REDHAWK_PALAPI PalGetTickCount()
 REDHAWK_PALEXPORT HANDLE REDHAWK_PALAPI PalGetModuleHandleFromPointer(_In_ void* pointer)
 {
     HANDLE moduleHandle = NULL;
+    // Emscripten's implementation of dladdr corrupts memory,
```

Comment:
This implementation won't lead to happiness in any place that calls into this. Did you look at the places where this is called from? Can we instead put `// WASMTODO: implement` there and ifdef the entire method out of here and out of the header?

The thing with workarounds like this is that eventually we'll hit all the places that call into this, rewrite them not to call this, but we'll still have a non-working `PalGetModuleHandleFromPointer` waiting for someone to get burned by. I kind of prefer ifdefing out the entire thing so that the runtime failure from someone calling it on a platform where they shouldn't becomes a compile time failure.

Reply:
I'm not quite sure what else to do. There are a number of usages of this, so it would be pretty hard to ifdef them all out. For what it's worth, I think they all have null checks or would at least hit a null dereference in an obvious place. Do you have thoughts on how we could do better?

Reply:
> There are a number of usages of this, so it would be pretty hard to ifdef them all out

There are 2 usages (the rest is ProjectN or Windows specific):

```
src\Native\Bootstrap\main.cpp(320):    void * osModule = PalGetModuleHandleFromPointer((void*)&CORERT_ENTRYPOINT);
src\Native\Runtime\startup.cpp(300):    if (!InitDLL(PalGetModuleHandleFromPointer((void*)&RhInitialize)))
```

Both of these will need to ifdefed out for WebAssembly eventually.

The first one has to do with stackwalking and exception handling that will work pretty differently for WebAssembly.

And the second one is Windows specific. It is not right even for Unix. Once we get stresslog to work on Unix, it will need to be ifdefed out for Unix - it may take care of it for WebAssembly as well.

Reply:
Ifdefed out PalGetModuleHandleFromPointer and replaced the usages with null. We'll probably have to revisit those nulls eventually. I'm not sure it's cleaner, but if you two think it's the way to go, we can do it.

Reply:
> I'm not sure it's cleaner, but if you two think it's the way to go, we can do it.

I was primarily commenting on the fact that there are not that many places.

Looking at the current delta, I tend to aggree that returning NULL from PalGetModuleHandleFromPointer like what you had originally would look better, for now at least (one stubbed out implementation tends to be better than multiple ifdefs).

Reply:
Switched this back to returning null. We'll probably notice via AVs if those other paths light up and start worrying about them then.